docs/maven-tutorial.md

```diff
@@ -0,0 +1,251 @@
+      <!--all requests to nexus via the mirror -->
+      <repositories>
+        <repository>
+          <id>central</id>
+          <url>http://central</url>
+          <releases><enabled>true</enabled></releases>
+          <snapshots><enabled>true</enabled></snapshots>
+        </repository>
+      </repositories>
+     <pluginRepositories>
+        <pluginRepository>
+          <id>central</id>
+          <url>http://central</url>
+          <releases><enabled>true</enabled></releases>
+          <snapshots><enabled>true</enabled></snapshots>
+        </pluginRepository>
+      </pluginRepositories>
+    </profile>
+  </profiles>
+  <activeProfiles>
+    <!--make the profile active all the time -->
+    <activeProfile>nexus</activeProfile>
+  </activeProfiles>
+</settings>
+```
+
+This file is telling Maven to use only the repository at
+`nexus.<Nexus_Project>:8081`, which is our proxy. Replace `<Nexus_Project>` with
+the project that has the Nexus repo. If the Nexus repo is in the same project as
+the applications using it, you can leave it out. More information on DNS
+resolution in OpenShift can be found
+[here](https://docs.openshift.org/latest/architecture/additional_concepts/networking.html).
+
+Create a Dockerfile in the wildfly-builder directory with a `FROM` entry that
+points to the builder image you normally use. Add a `COPY` entry pointing to
+the `settings.xml` you created earlier.
+
+```
+FROM openshift/wildfly-100-centos7:latest
+
+COPY settings.xml /usr/share/maven/conf/settings.xml
+COPY settings.xml /opt/app-root/src/.m2/settings.xml
+```
+
+> NOTE: Maven's default global settings are in `/usr/share/maven/conf`, but some
+> images specify a settings.xml file when it is called. For example, our WildFly
+> builder image uses `mvn -S /opt/app-root/src/.m2/settings.xml`, so we
+> overwrite that file with our own. You may have to do the same. In addition,
+> your image may have it's own global settings.xml file. In this case, you can
+> find the contents of this file, and integrate these settings into it. Use this
+> modified settings.xml file to overwrite the original.
+
+From your project's base directory, build a new imagestream from the
+'wildfly-builder' directory.
+```bash
+oc new-build wildfly-builder/ --to 'wildfly-nexus-10'
+oc start-build wildfly-nexus-10 --from-dir wildfly-builder/ -F
+```
+
+Wait for this build to finish. Afterward, start up your app with the modified
+image.
+
+```bash
+oc new-app wildfly-nexus-10:latest~https://github.com/bparees/openshift-jee-sample.git
+```
+
+#### Confirming Success
+
+In your web browser, navigate to
+`http://<Nexus IP>:8081/content/groups/public` to confirm that it has stored
+your application's dependencies. You can also check the build logs to see if
+Maven is using the Nexus mirror. If successful, you should see output
+referencing the URL `http://nexus:8081`.
+
+If you set up persistent storage, redeploy the Nexus server to make sure the
```

Comment:
Redeploying an app is not a good practice to check that the persistent storage is working. Just by going to the host's dir and check for the files should be enough.